src/subject_name/verify.rs

```diff
@@ -295,3 +300,148 @@ impl<'a> FromDer<'a> for GeneralName<'a> {
 
     const TYPE_ID: DerTypeId = DerTypeId::GeneralName;
 }
+
+#[cfg(feature = "alloc")]
+impl fmt::Debug for GeneralName<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            GeneralName::DnsName(name) => write!(
+                f,
+                "DnsName(\"{}\")",
+                String::from_utf8_lossy(name.as_slice_less_safe())
+            ),
+            GeneralName::DirectoryName => write!(f, "DirectoryName"),
+            GeneralName::IpAddress(ip) => {
+                write!(f, "IpAddress({:?})", IpAddrSlice(ip.as_slice_less_safe()))
+            }
+            GeneralName::UniformResourceIdentifier(uri) => write!(
+                f,
+                "UniformResourceIdentifier(\"{}\")",
+                String::from_utf8_lossy(uri.as_slice_less_safe())
+            ),
+            GeneralName::Unsupported(tag) => write!(f, "Unsupported({})", tag),
```

Comment:
Maybe slightly nicer to have the unsupported tag written in hex?

Reply:
Fair, done.